src/liquid/utils/scroll.ts

```diff
@@ -0,0 +1,23 @@
+export function getScrollParent(element: HTMLElement, includeHidden?: boolean) {
```

Comment:
I added some comments. Basically this:

> This is a vanilla JS port of the jQuery UI scrollParent method:
https://github.com/jquery/jquery-ui/blob/main/ui/scroll-parent.js

> Using for loop instead of recursion in order to save memory.